state/migration_import.go

```diff
@@ -617,6 +617,37 @@ func (i *importer) machinePortsOp(m description.Machine) txn.Op {
 	}
 }
 
+func (i *importer) applicationPortsOp(a description.Application) txn.Op {
+	modelUUID := i.st.ModelUUID()
+	appName := a.Name()
+	docID := i.st.docID(applicationGlobalKey(appName))
+
+	portRangeDoc := newApplicationPortRangesDoc(docID, modelUUID, appName)
+	for unitName, unitPorts := range a.OpenedPortRanges().ByUnit() {
+		portRangeDoc.UnitRanges[unitName] = make(network.GroupedPortRanges)
+
+		for endpointName, portRanges := range unitPorts.ByEndpoint() {
+			portRangeList := make([]network.PortRange, len(portRanges))
```

Comment:
`slice.Transform` from out collections package is nice for this boilerplate slice copying.

Reply:
Good to know this util func, ty!